packages/web3-providers-http/src/index.js

```diff
@@ -19,29 +19,33 @@
  *   Marek Kotewicz <marek@parity.io>
  *   Marian Oancea
  *   Fabian Vogelsteller <fabian@ethereum.org>
+ *   AyanamiTech <ayanami0330@protonmail.com>
  * @date 2015
  */
 
 var errors = require('web3-core-helpers').errors;
-var XHR2 = require('xhr2-cookies').XMLHttpRequest; // jshint ignore: line
+var fetch = require('cross-fetch');
 var http = require('http');
 var https = require('https');
 
+// Apply missing polyfill for IE
+require('es6-promise').polyfill();
+require('abortcontroller-polyfill/dist/polyfill-patch-fetch');
 
 /**
  * HttpProvider should be used to send rpc calls over http
  */
 var HttpProvider = function HttpProvider(host, options) {
     options = options || {};
 
-    this.withCredentials = options.withCredentials || false;
+    this.withCredentials = options.withCredentials;
```

Comment:
This is no longer boolean  https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#sending_a_request_with_credentials_included